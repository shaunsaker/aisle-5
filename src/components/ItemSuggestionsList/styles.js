import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    paddingLeft: styleConstants.dimensions.spacing.horizontal,
    paddingBottom: styleConstants.dimensions.spacing.vertical,
  },
  contentContainer: {
    paddingRight: 50 + 2 * styleConstants.dimensions.spacing.horizontal, // iconButton
  },
  itemContainer: {
    marginRight: styleConstants.dimensions.spacing.horizontal,
  },
});

export default styles;
