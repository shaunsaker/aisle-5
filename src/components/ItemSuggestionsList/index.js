import React from 'react';
import PropTypes from 'prop-types';
import { FlatList } from 'react-native';
import Animator from 'react-native-simple-animators';

import styleConstants from '../../styleConstants';

import styles from './styles';

import Label from '../Label';

export default class ItemSuggestionsList extends React.Component {
  constructor(props) {
    super(props);

    this.renderItem = this.renderItem.bind(this);
  }

  static propTypes = {
    data: PropTypes.arrayOf(PropTypes.any),
    handlePress: PropTypes.func,
  };

  static defaultProps = {};

  renderItem({ item, index }) {
    const { handlePress } = this.props;

    return (
      <Animator
        type="translateY"
        initialValue={32 + styleConstants.dimensions.spacing.vertical}
        finalValue={0}
        shouldAnimateIn
        delay={index * 100}
        style={styles.itemContainer}
      >
        <Animator
          type="scale"
          initialValue={0.5}
          finalValue={1}
          shouldAnimateIn
          delay={index * 100}
        >
          <Label text={item.name} handlePress={() => handlePress(item.name)} />
        </Animator>
      </Animator>
    );
  }

  render() {
    const { data } = this.props;

    return (
      <FlatList
        keyExtractor={({ id }) => id}
        data={data}
        renderItem={this.renderItem}
        style={styles.container}
        contentContainerStyle={styles.contentContainer}
        horizontal
      />
    );
  }
}
