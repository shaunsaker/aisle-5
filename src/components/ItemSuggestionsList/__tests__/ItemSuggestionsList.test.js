import React from 'react';
import renderer from 'react-test-renderer';

import ItemSuggestionsList from '..';
import USER_ITEM from '../../../mockData/userItem';

// Fixes _bezier is not a function bug
jest.useFakeTimers();

describe('ItemSuggestionsList', () => {
  const spies = [];
  const dispatch = jest.fn();
  const data = [USER_ITEM];

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<ItemSuggestionsList data={data} />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle renderItem', () => {
      // NOTE: We assume this works and is handled by RN FlatList
    });
  });

  describe('actions', () => {
    // NOTE: We assume simple prop actions work here
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
    dispatch.mockClear();
  });
});
