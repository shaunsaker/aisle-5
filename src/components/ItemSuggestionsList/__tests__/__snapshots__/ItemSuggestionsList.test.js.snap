// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemSuggestionsList renders renders with minimum required props 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingRight": 65,
    }
  }
  data={
    Array [
      Object {
        "id": "1",
        "name": "Beer",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  initialNumToRender={10}
  keyExtractor={[Function]}
  keyboardShouldPersistTaps="always"
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsHorizontalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  style={Object {}}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 42,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 7.5,
              "transform": Array [
                Object {
                  "scale": 0.5,
                },
              ],
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": null,
                "borderRadius": 5,
                "borderWidth": 0,
                "elevation": 6,
                "opacity": 1,
                "paddingHorizontal": 7.5,
                "paddingVertical": 5,
                "shadowColor": "#212121",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.33,
                "shadowRadius": 2,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#08293A",
                  "fontFamily": "Muli-SemiBold",
                  "fontSize": 17,
                }
              }
            >
              Beer
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
